README.md

```diff
@@ -249,22 +249,24 @@ This application defines Docker images for production, testing, and development
 
 ---
 
 ## Deployment Workflow
+The Docker image generated and used in local development with `ci/Dockerfile.local`, builds with the Python packages given by `requirements.txt`.  This ensures that the local image is built with the same dependencies used in the deployed environments.
 
-### Update requirements.txt
+The Docker image used in deployed environments, `ci/Dockerfile`, builds with the Python packages given by `requirements.txt`, which uses frozen dependencies, which can be updated as described in the [Update Requirements](#update-requirements).
 
-The Docker image used in local development, ci/Dockerfile.local, builds with the Python packages given by requirements-app.txt, which specifies only the top level dependencies.  This ensures that the local image always builds with the most recent sub-dependencies, and developers only need to keep track of the top level dependencies.
+### Update requirements
+Updating `requirements.txt` is done by first building locally, using `requirements-app.txt`, which contains all top-level dependencies. The full process to upgrade is as follows:
 
-The Docker image used in production, ci/Dockerfile, builds with the Python packages given by requirements.txt, which freezes all dependencies.  **Prior to any deployment to AWS, update requirements.txt as follows**:
-
-1. If necessary, build the notification_api Docker image using the docker-compose command given in [Local Development](#local-development).
-2. Run `docker run --rm -i notification_api pip freeze > requirements.txt`.
-3. Open requirements.txt, and manually remove any warning messages at the start of the file.
-4. Assuming all unit tests are passing, note any top level dependency updates.  Update requirements-app.txt to make their minimum version equal to the version actually installed according to requirements.txt.
+1. In `ci/Dockerfile.local`, replace all instances of `requirements.txt` with `requirements-app.txt`
+2. Build the notification_api Docker image using the `docker compose` command given in [Local Development](#local-development).
+3. Run `docker run --rm -i notification_api pip freeze > requirements.txt`.
+4. Open `requirements.txt`, and manually remove any warning messages at the start of the file.
+5. Assuming all unit tests are passing, note any top level dependency updates in requirements.txt.  Update `requirements-app.txt` to make their minimum version equal to the version actually installed according to `requirements.txt`.
+6. Replace `requirements-app.txt` with `requirements.txt` in `ci/Dockerfile.local` (undo step 1)
```

Comment:
I suggest reverting these changes and rebasing this branch to master, which has newer instructions.